### PR TITLE
Forbid to have an extension method in the package owning the class

### PR DESCRIPTION
**Pull request by jecisc**
It is currently possible to have a method that is an extension method that is in the same package as its class. This creates weird behaviors were some tools consider it an extension method and some other consider it not an extension method.

To make thing more coherent I propose that we forbid to have this by classifying such methods under the unclassified protocol instead.

Fixes ##14353

### CODE REVIEW

File: src/Kernel/ClassDescription.class.st [hunk 1]
```diff
@@ -264,6 +264,9 @@ ClassDescription >> classify: selector under: aProtocol [
 
 	"In case the method is already classified and we say to classify it under nil, we do not update the protocol."
 	(aProtocol isNil and: [ oldProtocol isNotNil ]) ifTrue: [ ^ self ].
+	
+	"If we try to classify the method as an extension of its own package, just unclassify the method to avoid any weird state where the method could be considered as an extension or not depending on the interpretation."
+	(self isProtocolExtensionFromTheSamePackage: aProtocol) ifTrue: [ ^ self classify: selector under: Protocol unclassified ].
```
Comment (tesonep) [inline src/Kernel/ClassDescription.class.st]: Why not also signalling a warning to notify this problem? 
The warning can proceed in interactive mode and is logged in headless mode.

File: src/Kernel/ClassDescription.class.st [hunk 2]
```diff
@@ -959,6 +979,9 @@ ClassDescription >> renameProtocol: anOldProtocol as: aNewProtocol [
 	(aNewProtocol isNil or: [ anOldProtocol isNil ]) ifTrue: [ ^ self ].
 
 	(self hasProtocol: anOldProtocol) ifFalse: [ ^ self ].
+	
+	"We should not be able to have a protocol that is an extension of the same class so let's skip the rename."
+	(self isProtocolExtensionFromTheSamePackage: aNewProtocol) ifTrue: [ ^ self ].
```
Comment (tesonep) [inline src/Kernel/ClassDescription.class.st]: The same as before, it is better to generate a warning.

Comment (tesonep) [reply src/Kernel/ClassDescription.class.st]: Having something that does nothing without telling the user is really bad behaviour.

**Review by tesonep — CHANGES_REQUESTED** [review]

File: src/Kernel/ClassDescription.class.st (hunk 1 above)
Comment (jecisc) [reply src/Kernel/ClassDescription.class.st]: I hesitated about that but I agree that it is better. I'll update the PR

**Comment by jecisc** [comment]
Apparently the warning is preventing the bootstrap to finish. Do you know what I should do so that it just log the problem and proceed?

**Comment by jecisc** [comment]
Checking the code of the compiler it seems that the right way would be to use a Notification or even a SystemNotification instead of a warning

**Comment by jecisc** [comment]
The build now passes and things are working correctly. In the non interactive environment things get logged. 
I just broke some tests that I'll fix now

**Comment by jecisc** [comment]
The last failure is fixed by this: https://github.com/pillar-markup/Microdown/pull/658

But I need someone with merge rights :'(

**Comment by jecisc** [comment]
The requested changes were applied and the tests are now green

**Pull request merged**